src/feat/pitch-functions.cc

```diff
@@ -856,14 +857,21 @@ void OnlinePitchFeatureImpl::ExtractFrame(
   } else {
     // frame is partly in the remainder and partly in the new part.
     int32 remainder_offset = downsampled_signal_remainder_.Dim() + offset;
-    KALDI_ASSERT(remainder_offset >= 0);  // or we didn't keep enough remainder.
+    /*KALDI_ASSERT(remainder_offset >= 0);*/  // or we didn't keep enough remainder.
     KALDI_ASSERT(offset + full_frame_length > 0);  // or we should have
                                                    // processed this frame last
                                                    // time.
 
     int32 old_length = -offset, new_length = offset + full_frame_length;
-    window->Range(0, old_length).CopyFromVec(
-        downsampled_signal_remainder_.Range(remainder_offset, old_length));
+
+    // BP: this looks to me like something has not been done properly
```

Comment:
You are right. I have now refactored the code a bit and I think the latest commit make things more clear, as the two edge cases are now treated in a similar manner.